Visual Studio(VCC) 2019 cannot compile empty tuple generated by Nim

It works with gcc.

### Command line

```bash
nim c -r --cc:vcc bugtup
```

### Example
```nim
var bugTup = ()

proc main() =
  debugEcho bugTup
  
main()
```

### Current Output
```
C:\Users\AndriLim\nimcache\bugtup_d\@mbugtup.nim.c(64): error C2059: syntax error: '}'
Error: execution of an external compiler program 'cl.exe /c /nologo   /IF:\projects\status-nim\lib /IF:\projects\bug /FoC:\Users\AndriLim\nimcache\bugtup_d\@mbugtup.nim.c.obj C:\Users\AndriLim\nimcache\bugtup_d\@mbugtup.nim.c' failed with exit code: 2
```

### Expected Output
```
()
```

The compiler produce this C code and VCC 2019 seems doesn't like the `{}`.

```C
N_LIB_PRIVATE tyTuple__qERWGavQjzug69cyzEYP39bQ bugTup__XaG7QrFzC9aZhBgUBoCX3SQ = {};
```

```
$ nim -v
Nim Compiler Version 1.3.5 [Windows: amd64]
Compiled at 2020-06-10
Copyright (c) 2006-2020 by Andreas Rumpf

git hash: 8bbdb8f43fb0da04d4b75bca011662f690972bfe
active boot switches: -d:release
```

--- a/compiler/ccgexprs.nim
+++ b/compiler/ccgexprs.nim
@@ -3200,6 +3200,8 @@ proc getDefaultValue(p: BProc; typ: PType; info: TLineInfo; result: var Rope) =
     result.add "}"
   of tyTuple:
     result.add "{"
+    if p.vccAndC and t.len == 0:
+      result.add "0"
     for i in 0..<t.len:
       if i > 0: result.add ", "
       getDefaultValue(p, t[i], info, result)
@@ -3338,6 +3340,8 @@ proc genConstSimpleList(p: BProc, n: PNode; isConst: bool; result: var Rope) =
 
 proc genConstTuple(p: BProc, n: PNode; isConst: bool; tup: PType; result: var Rope) =
   result.add "{"
+  if p.vccAndC and n.len == 0:
+    result.add "0"
   for i in 0..<n.len:
     let it = n[i]
     if i > 0: result.add ",\n"

--- a/compiler/cgendata.nim
+++ b/compiler/cgendata.nim
@@ -172,6 +172,7 @@ type
 
 template config*(m: BModule): ConfigRef = m.g.config
 template config*(p: BProc): ConfigRef = p.module.g.config
+template vccAndC*(p: BProc): bool = p.module.config.cCompiler == ccVcc and p.module.config.backend == backendC
 
 proc includeHeader*(this: BModule; header: string) =
   if not this.headerFiles.contains header:

--- a/tests/misc/t14667.nim
+++ b/tests/misc/t14667.nim
@@ -1,0 +1,12 @@
+discard """
+  matrix: "--cc:vcc"
+  disabled: "linux"
+  disabled: "bsd"
+  disabled: "osx"
+  disabled: "unix"
+  disabled: "posix"
+"""
+
+type A = tuple
+discard ()
+discard default(A)

